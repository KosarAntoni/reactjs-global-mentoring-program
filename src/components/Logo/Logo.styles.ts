import styled from '@emotion/styled'

export const StyledLogo = styled.span`
  font-size: 2rem;
  color: var(--pastele-red);

  b {
    font-weight: 900;
  }
`
