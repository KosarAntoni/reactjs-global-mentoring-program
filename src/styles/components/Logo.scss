.logo {
  font-size: 2rem;
  color: var(--pastele-red);

  b {
    font-weight: 900;
  }
}
