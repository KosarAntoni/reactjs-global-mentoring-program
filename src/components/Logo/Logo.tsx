import React, { FC } from 'react'
import classNames from 'classnames'

import { LogoProps } from './Logo.models'

const Logo: FC<LogoProps> = ({ className }) => (
  <span className={classNames('logo', className)}>
    <b>netflix</b>
    roulette
  </span>
)

export default Logo
