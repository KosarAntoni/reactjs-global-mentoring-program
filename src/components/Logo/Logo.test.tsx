import React from 'react'
import { render, screen } from '@testing-library/react'

import Logo from './Logo'

import '@testing-library/jest-dom'

test('the Logo component renders correctly with required and optional props', () => {
  render(<Logo className='test-class' />)

  const node = screen.getByTestId('logo-node')

  // the component has been rendered
  expect(node).toBeInTheDocument()

  // generating a snapshot, not for testing purposes but only to see what changes in the HTML structure during CR if the developer updates the snapshot
  expect(node).toMatchSnapshot()

  // the comonent has correct classes
  expect(node).toHaveClass('logo test-class')
})
