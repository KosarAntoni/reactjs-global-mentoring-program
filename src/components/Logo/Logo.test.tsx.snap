// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Logo component renders correctly with required and optional props 1`] = `
<span
  class="logo test-class"
  data-testid="logo-node"
>
  <b>
    netflix
  </b>
  roulette
</span>
`;
